src/CG/CG_elem.cpp

```diff
@@ -2113,6 +2113,829 @@ bool intersectSegmentPolygon( array3D const &P0, array3D const &P1, std::size_t
+            tup1  = std::next(ordered_intersections.begin(), iL);
+            tup2  = std::next(ordered_intersections.begin(), iR);
+            found = norm2(tup1->second - tup2->second) > distanceTolerance;
+        }
+
+        std::size_t offset = 0;
+        if (found) {
+            // check the normal of polygon portion cut by segment has the same normal direction than my original polygon.
+            // if not alter the offset so that i can visit intersection n-1,0,1,2,... 2 by 2, instead of 0,1,2,...,n-1
+            array3D polyNormal = computePolygonNormal(nV, V);
+
+            std::vector<array3D> pp;
+            pp.reserve(tup2->first - tup1->first + 2);
+            pp.push_back(tup1->second);
+            for (int k = (tup1->first + 1); k <= tup2->first; ++k) {
+                pp.push_back(V[k]);
+            }
+            pp.push_back(tup2->second);
+
+            array3D subNormal = computePolygonNormal(pp.size(), pp.data());
+            offset            = (dotProduct(polyNormal, subNormal) < 0) ? size - 1 : 0;
+        }
+        for (std::size_t j = 0; j < (size / 2); ++j) {
+            tup1 = std::next(ordered_intersections.begin(), (2 * j + offset) % size);
+            tup2 = std::next(ordered_intersections.begin(), (2 * j + 1 + offset) % size);
+            Qs.push_back(std::array<array3D, 2>{tup1->second, tup2->second});
+            edges_of_Qs.push_back(std::array<int, 2>{tup1->first, tup2->first});
+        }
+    }
+    return !Qs.empty();
+}
+
+/*!
+ * Once a polygon intersects a plane and the intersection segments info are available, 
+ * return the N polygons created by the intersections on both sides of the plane.
+ * \param[in] nV Number  of vertices of the polygon
+ * \param[in] V Pointer to vertices list array. Must have size nV.
+ * \param[in] Qs List of cutting segments identified by the plane.
+ * \param[in] edges_of_Qs List of polygon edges couple "touched" by each cutting segment. Negative indices or edge local indices greater
+ * than number for actual polygon edges, will make the method skip the info.
+ * \param[out] polys List of resulting polygons. Each new polygon will be reported as a list of constitutive points. In case of invalid input, return a empty list. 
+ * \param[in] distanceTolerance if distance among features exceed this value they are considered as not intersecting
+ */
+void reconstructPlaneIntersectedPolygons(std::size_t nV, array3D const *V,
+                                         std::vector<std::array<array3D, 2>> const &Qs,
+                                         std::vector<std::array<int, 2>> const &edges_of_Qs,
+                                         std::vector< std::vector<array3D> > &polys,
+                                         const double distanceTolerance)
+{
+    polys.clear();
+
+    std::vector<array3D> motherPolygon;
+    std::size_t motherPos(0);
+    int motherInit(-1);
+    motherPolygon.reserve(nV);
+
+    std::size_t size = Qs.size();
```

Comment:
Please add some comments explaining what motherPolygon is.

Reply:
Done